slick/src/main/scala/slick/jdbc/PostgresProfile.scala

```diff
@@ -255,12 +258,191 @@ trait PostgresProfile extends JdbcProfile {
+
+      implicit val min : LocalTime = LocalTime.MIN
+      implicit val max : LocalTime = LocalTime.MAX
+      implicit val serializeFiniteTime : (LocalTime => String) =  _.format(formatter)
+      implicit val parseFiniteTime : (String => LocalTime) = LocalTime.parse(_, formatter)
+
+      override val sqlType = java.sql.Types.OTHER
+      override def sqlTypeName(sym: Option[FieldSymbol]) = "TIME"
+      override def setValue(v: LocalTime, p: PreparedStatement, idx: Int) = {
+        p.setObject(idx, serializeTime(v), sqlType)
+      }
+      override def updateValue(v: LocalTime, r: ResultSet, idx: Int) = {
+        r.updateObject(idx, createPGObject(serializeTime(v), sqlTypeName(None)))
+      }
+      override def getValue(r: ResultSet, idx: Int): LocalTime = parseTime(r.getString(idx))
+      override val hasLiteralForm : Boolean = false
+    }
+
+    class OffsetTimeJdbcType extends super.OffsetTimeJdbcType with PostgreTimeJdbcType[OffsetTime] {
+
+      private[this] val formatter : DateTimeFormatter = {
+        new DateTimeFormatterBuilder()
+          .append(DateTimeFormatter.ofPattern("HH:mm:ss"))
+          .optionalStart()
+          .appendFraction(ChronoField.NANO_OF_SECOND, 0, 6, true)
+          .optionalEnd()
+          .appendOffset("+HH:mm", "+00")
+          .toFormatter()
+      }
+
+      implicit val min : OffsetTime = OffsetTime.MIN
+      implicit val max : OffsetTime = OffsetTime.MAX
+      implicit val serializeFiniteTime : (OffsetTime => String) =  _.format(formatter)
+      implicit val parseFiniteTime : (String => OffsetTime) = OffsetTime.parse(_, formatter)
+
+      override val sqlType = java.sql.Types.OTHER
+      override def sqlTypeName(sym: Option[FieldSymbol]) = "TIMETZ"
+      override def setValue(v: OffsetTime, p: PreparedStatement, idx: Int) = {
+        p.setObject(idx, serializeTime(v), sqlType)
+      }
+      override def updateValue(v: OffsetTime, r: ResultSet, idx: Int) = {
+        r.updateObject(idx, createPGObject(serializeTime(v), sqlTypeName(None)))
+      }
+      override def getValue(r: ResultSet, idx: Int): OffsetTime = parseTime(r.getString(idx))
+      override val hasLiteralForm : Boolean = false
+    }
+
+    class InstantJdbcType extends super.InstantJdbcType with PostgreTimeJdbcType[Instant] {
+
+      private[this] val formatter = {
+        new DateTimeFormatterBuilder()
+          .append(DateTimeFormatter.ofPattern("yyyy-MM-dd HH:mm:ss"))
+          .optionalStart()
+          .appendFraction(ChronoField.NANO_OF_SECOND, 0, 6, true)
+          .optionalEnd()
+          .optionalStart()
+          .appendLiteral("+00")
+          .optionalEnd()
+          .toFormatter()
+      }
+
+      implicit val min : Instant = Instant.MIN
+      implicit val max : Instant = Instant.MAX
+      implicit val serializeFiniteTime : (Instant => String) =  _.toString
+      implicit val parseFiniteTime : (String => Instant) = {
+        LocalDateTime.parse(_, formatter).toInstant(ZoneOffset.UTC)
```

Comment:
Is this really safe? Here you convert the string from the database to the Local timezone and then to Instant. As I understand it, you would get wrong time if the system timezone is not UTC and the time didn't contain a timezone (which would be OK for instant since it is implicitly always in UTC)